sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -301,10 +307,40 @@ case class StringLength(child: Expression) extends UnaryExpression with ExpectsI
 }
 
 /**
- * A function that return the Levenshtein distance between the two given strings.
+ * Returns string, with the first letter of each word in uppercase,
+ * all other letters in lowercase. Words are delimited by whitespace.
  */
+case class InitCap(child: Expression) extends UnaryExpression with ExpectsInputTypes {
+  override def dataType: DataType = StringType
+
+  override def inputTypes: Seq[DataType] = Seq(StringType)
+
+  override def eval(input: InternalRow): Any = {
+    val string = child.eval(input)
+    if (string == null) {
+      null
+    }
+    else if (string.asInstanceOf[UTF8String].length == 0) {
+      return UTF8String.fromString(string.toString)
+    }
+    else {
+      val str: String = string.toString
+      val sb = new StringBuilder
+      sb.append(str)
+      sb.setCharAt(0, sb(0).toUpper)
+
+      for (i <- 1 until sb.length) {
+        if (sb(i - 1).equals(' ')) {
```

Comment:
@davies Should `initCap` consider only spaces or all whitespaces: `Character.isWhitespace`, see http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/8u40-b25/java/lang/Character.java#Character.isWhitespace%28char%29